main.tf

```diff
@@ -240,6 +270,18 @@ resource "aws_cloudfront_distribution" "default" {
     }
   }
 
+  dynamic "origin" {
+    for_each = var.s3_origins
```

Comment:
what if both `var.custom_origins` and `var.s3_origins` are provided ? can a cloudfront distribution have both ? or are they mutually exclusive ?

Reply:
An S3 origin is for S3 buckets, and custom origins are for S3 websites or apps. Both can be used at the same time. I use both.